### PR TITLE
(refactor): org-roam-db--clear-file pull tables from schemata

### PR DESCRIPTION
**Pull request by progfolio**
Programatically pull tables from schemata, rather than manually
specifying.

### CODE REVIEW

File: org-roam-db.el [hunk 1]
```diff
@@ -192,25 +192,14 @@ the current `org-roam-directory'."
...
-    (org-roam-db-query [:delete :from links
-                        :where (= from $s1)]
-                       file)
-    (org-roam-db-query [:delete :from titles
-                        :where (= file $s1)]
-                       file)
-    (org-roam-db-query [:delete :from refs
-                        :where (= file $s1)]
-                       file)))
+  (let ((file (file-truename (or filepath (buffer-file-name)))))
```
Comment (zaeph) [inline org-roam-db.el:198]: Maybe we need to include indirect buffers as well.
```suggestion
  (let ((file (file-truename (or filepath
                                 (buffer-file-name)
                                 (buffer-file-name (buffer-base-buffer))))))
```

Comment (progfolio) [reply org-roam-db.el:198]: Good catch. No need for the first call to `(buffer-file-name)` though. `(buffer-base-buffer)` will return nil if there is no base-buffer, which will result in current-buffer being used in `(buffer-file-name)`.

Comment (progfolio) [reply org-roam-db.el:198]: `(buffer-file-name (buffer-base-buffer))` is correct for both cases.

Comment (zaeph) [reply org-roam-db.el:198]: Oh yeah, you're right.  I'm pretty sure I did exactly the same mistake in a previous PR.

**Pull request merged**